addons/hr_holidays/models/hr_leave.py

```diff
@@ -1022,8 +1064,11 @@ def action_validate(self):
         self.filtered(lambda holiday: holiday.validation_type == 'both').write({'second_approver_id': current_employee.id})
         self.filtered(lambda holiday: holiday.validation_type != 'both').write({'first_approver_id': current_employee.id})
 
-        for holiday in self.filtered(lambda holiday: holiday.holiday_type != 'employee'):
-            if holiday.holiday_type == 'category':
+        for holiday in self.filtered(lambda holiday: holiday.holiday_type != 'employee' or\
```

Comment:
We are calling 3 times filtered on the same recordset... Couldn't we do a single loop and making the checks in the loop?